hooks/shapediver/viewer/interaction/useSelectManagerEvents.ts

```diff
@@ -0,0 +1,133 @@
+import { notifications } from "@mantine/notifications";
+import { addListener, EVENTTYPE_INTERACTION, IEvent, removeListener } from "@shapediver/viewer";
+import { InteractionEventResponseMapping, MultiSelectManager } from "@shapediver/viewer.features.interaction";
+import { useState, useCallback, useEffect } from "react";
+import { processNodes } from "./utils/patternUtils";
+
+// #region Functions (1)
+
+/**
+ * Hook allowing to create the select manager events.
+ * 
+ * @param viewportId 
+ */
+export function useSelectManagerEvents(pattern: { [key: string]: string[][] }): {
+	/**
+	 * The selected node names.
+	 */
+    selectedNodeNames: string[],
+	/**
+	 * Callback function to reset the selected node names.
+	 * 
+	 * @returns 
+	 */
+    resetSelectedNodeNames: () => void
+} {
+	// state for the selected nodes
+	const [selectedNodeNames, setSelectedNodeNames] = useState<string[]>([]);
+	const resetSelectedNodeNames = useCallback(() => setSelectedNodeNames([]), []);
+
+	// register an event handler and listen for output updates
+	useEffect(() => {
+		/**
+         * Event handler for the select on event.
+         * In this event handler, the response object is created and the parameter is updated.
+         */
+		const tokenSelectOn = addListener(EVENTTYPE_INTERACTION.SELECT_ON, async (event: IEvent) => {
+			const selectEvent = event as InteractionEventResponseMapping[EVENTTYPE_INTERACTION.SELECT_ON];
+
+			// don't send the customization if the event is coming from an API call
```

Comment:
@MajorMeerkatThe3rd this comment and others seem a bit misleading to me, maybe better: _We ignore the event if it's not based on an event triggered by the UI._ (or similar).